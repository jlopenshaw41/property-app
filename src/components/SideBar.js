import React from "react";
import { Link } from "react-router-dom";

const SideBar = () => {
  return (
    <div>
      <Link className="link" to={`/?query={"city": "Manchester"}`} />
      <Link className="link" to={`/?query={"city": "Leeds"}`} />
      <Link className="link" to={`/?query={"city": "Sheffield"}`} />
      <Link className="link" to={`/?query={"city": "Liverpool"}`} />
    </div>
  );
};

export default SideBar;
