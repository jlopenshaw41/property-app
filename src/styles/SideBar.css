.sidebar {
    display: flex;
    flex-direction: column;
    padding: 10px;
    border-right: 1px solid gray; 
}

.search-form {
    padding: 5px;
}

.link {
    padding: 5px;
}

.sidebar p {
    padding: 5px;
    font-weight: bold;
}