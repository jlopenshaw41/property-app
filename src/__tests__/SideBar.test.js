import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import SideBar from "../components/SideBar";

describe("SideBar", () => {
  it("renders as expected", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router history={history}>
        <SideBar />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
