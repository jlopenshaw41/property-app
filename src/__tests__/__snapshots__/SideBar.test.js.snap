// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders as expected 1`] = `
<DocumentFragment>
  <div>
    <a
      class="link"
      href="/?query={\\"city\\": \\"Manchester\\"}"
    />
    <a
      class="link"
      href="/?query={\\"city\\": \\"Leeds\\"}"
    />
    <a
      class="link"
      href="/?query={\\"city\\": \\"Sheffield\\"}"
    />
    <a
      class="link"
      href="/?query={\\"city\\": \\"Liverpool\\"}"
    />
  </div>
</DocumentFragment>
`;
